src/uri-helper.ts

```diff
@@ -0,0 +1,45 @@
+import qs from 'qs'
+
+export interface HandlerOpiton {
+  user?: string
+  base?: string
+}
+
+export interface HandlerQuery {
+  uri?: string
+  limit?: number
+  offset?: number
+  query?: any
+  [key: string]: any
+}
+
+export class UriHelper {
+  public baseUri: string
+
+  constructor(endpoint: string, options: HandlerOpiton) {
+    const base = options.base || 'https://api.tillhub.com'
+    const user = options.user ? `/${options.user}` : ''
+    this.baseUri = `${base}${endpoint}${user}`
+  }
+
+  public generateBaseUri(path?: string): string {
+    const additionalPath = path || ''
+    return `${this.baseUri}${additionalPath}`
+  }
+
+  public generateUriWithQuery(basePath: string, query?: HandlerQuery): string {
+    let uri
+    if (!query) {
+      uri = `${basePath}`
+    } else if (query.uri || (query.query && query.query.uri)) {
+      uri = query.uri || query.query.uri
+    } else if (query.query) {
+      let flattenQuery = { ...query, ...query.query }
```

Comment:
Should we remove the "uri" query param? (also further down below)

Reply:
I think the it's unnecessary since the uri is taken care of in the second if conditions.